### PR TITLE
Add new certified category views

### PR DESCRIPTION
**Pull request by carkod**
## Done

Added views for each category
Refactored into a single function that dynamically generates the different category search results

## QA

- Check https://ubuntu-com-10114.demos.haus/certified/desktops
- Check https://ubuntu-com-10114.demos.haus/certified/laptops
- Check https://ubuntu-com-10114.demos.haus/certified/socs
- Check https://ubuntu-com-10114.demos.haus/certified/devices
- Check https://ubuntu-com-10114.demos.haus/certified/servers
Or you can also go to the top navigation and try all the links, they should also work.

## Issue / Card

Fixes https://github.com/canonical-web-and-design/web-squad/issues/4286
Fixes https://github.com/canonical-web-and-design/web-squad/issues/4287
Fixes https://github.com/canonical-web-and-design/web-squad/issues/4288
Fixes https://github.com/canonical-web-and-design/web-squad/issues/4289

### CODE REVIEW

**Comment by bethcollins92** [comment]
I'll review this once we've merged mine so we won't need all that javascript on each page 🥳

**Review by bethcollins92 — APPROVED** [review]
LGTM 👍🏼 . Couple of points:

- I'll refactor this with an include for the repeated code as you mentioned, which I can do when I add content to these pages [here](https://github.com/canonical-web-and-design/web-squad/issues/4290).
-  @davegoddard42 would you agree that when we're on the desktop specific page for example, the list of vendors and releases that are displayed in the filter section should only include ones which come under the desktop category? I feel we shouldn't have them in the list if there aren't any desktops by that vendor/release. (Out of scope for this PR but can make another issue)

**Comment by davegoddard42** [comment]
@bethcollins92 yes, that would make sense.

**Comment by carkod** [comment]
> @bethcollins92 yes, that would make sense.

Created issue https://github.com/canonical-web-and-design/web-squad/issues/4317

File: templates/certified/desktops.html [hunk 1]
```diff
@@ -0,0 +1,189 @@
+{% extends "certified/base_certification.html" %}
+
+{% block title %}Certified hardware{% endblock %}
```
Comment (anthonydillon) [inline templates/certified/desktops.html:3]: Could this be "Certified desktops"

File: templates/certified/desktops.html [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+{% block meta_description %}Desktops that have been certified for use with Ubuntu.{% endblock meta_description %}
```
Comment (anthonydillon) [inline templates/certified/desktops.html:5]: Are there copy docs for these pages? I feel a better description would be something like: "Ubuntu certifies thousands of desktops to guarantee they work well with Ubuntu." Gives more of a purpose of the page and content.

File: templates/certified/desktops.html [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+              {% endif %}
+              <option value="{{ total_results }}" name="all">All</option>
+            </select>
+            {% endif %}
+          </div>
+        </div>
+      </div>
+      {% else %}
+      <div class="col-9">
+        <h3 class="p-heading--4">No results - why not try widening your search?</h3>
```
Comment (anthonydillon) [inline templates/certified/desktops.html:170]: Maybe "Sorry we found on results matching your criteria. Please try and widen your search."

File: templates/certified/desktops.html [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        <p>You can do this by:</p>
+        <ul class="p-list">
+          <li class="p-list__item is-ticked u-sv1">
+            Adding alternative words or phrases
+          </li>
+          <li class="p-list__item is-ticked  u-sv1">
+            Using individual words instead of phrases
```
Comment (anthonydillon) [inline templates/certified/desktops.html:177]: `p` tags used within the `lil` will give you the spacing you are looking for without `u-sv1`

File: templates/certified/devices.html [hunk 5]
```diff
@@ -0,0 +1,189 @@
+{% extends "certified/base_certification.html" %}
+
+{% block title %}Certified hardware{% endblock %}
```
Comment (anthonydillon) [inline templates/certified/devices.html:3]: Same comments as the above document.

File: templates/certified/devices.html [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+{% block meta_description %}Devices that have been certified for use with Ubuntu.{% endblock meta_description %}
```
Comment (anthonydillon) [inline templates/certified/devices.html:5]: Try and find some wording that defines the purpose of the content.

File: templates/certified/laptops.html [hunk 7]
```diff
@@ -0,0 +1,189 @@
+{% extends "certified/base_certification.html" %}
+
+{% block title %}Certified hardware{% endblock %}
+
+{% block meta_description %}Laptops that have been certified for use with Ubuntu.{% endblock meta_description %}
```
Comment (anthonydillon) [inline templates/certified/laptops.html:5]: Same here

File: templates/certified/servers.html [hunk 8]
```diff
@@ -0,0 +1,189 @@
+{% extends "certified/base_certification.html" %}
+
+{% block title %}Certified hardware{% endblock %}
+
+{% block meta_description %}Servers that have been certified for use with Ubuntu.{% endblock meta_description %}
```
Comment (anthonydillon) [inline templates/certified/servers.html:5]: Same here

File: templates/certified/socs.html [hunk 9]
```diff
@@ -0,0 +1,189 @@
+{% extends "certified/base_certification.html" %}
+
+{% block title %}Certified hardware{% endblock %}
+
+{% block meta_description %}SoCs that have been certified for use with Ubuntu.{% endblock meta_description %}
```
Comment (anthonydillon) [inline templates/certified/socs.html:5]: Same here

File: templates/certified/desktops.html (hunk 2 above)
Comment (bethcollins92) [reply templates/certified/desktops.html:5]: FYI Dave is still chasing the PM for copy for these pages. There are rough drafts [here](https://drive.google.com/drive/folders/1a_Iz5iw7nt4VeInmHoACfCihW7ZU83bI) but they need looking at again and also need copy for the vendor specific pages

Comment (davegoddard42) [reply templates/certified/desktops.html:5]: I am actively chasing - spoke to the PM today. I've created a template for the vendor page too

File: templates/certified/devices.html (hunk 6 above)
Comment (carkod) [reply templates/certified/devices.html:5]: I would rather leave this to the copy docs / PM, whatever I come up with will most likely be replaced soon.

**Pull request merged**